ui/admin/app/components/form/field/secret-input/index.hbs

```diff
@@ -3,7 +3,7 @@
   SPDX-License-Identifier: MPL-2.0
 }}
 
-<div class='secret-input'>
+<div class='secret-input' data-test={{@data-test-selector}}>
```

Comment:
@ZedLi @cameronperera 

(This is just a suggested approach, totally open to discussion)

Why adding this to the component?

If we assign a `data-test-*` where we define the `<Form::Field::SecretInput>` the data test selector gets applied only to the input, leaving us with no specific way to select and interact with other elements of this component, such as buttons.

I come with this approach, where the user should assign a `@data-test-selector` attribute to the SecretInput component. So at rendering time, the wrapper for the SecretInput component will have a `data-test-*` as follows: `data-test=access_key_id`. Therefore we can select the button within the component as follows 
[data-test=access_key_id] [type=button].

Be aware, if you do not provide a `@data-test-selector` attribute when using the SecretInput, nothing related to test selectors will be rendered.

Reply:
An alternative approach:

The consumer of this component does NOT provide the attribute `@data-test-selector` and we use the attribute `@name` as value for data-test. i.e: `data-test={{@name}}`
This is simpler since we provide `@name` for all the fields.

Reply:
I like the alternative approach, since I think most likely they're going to end up being the same

Reply:
@ZedLi Awesome, yeah me too, I just realize about the second approach later. I will apply it.